File: RoRServer/src/main/java/commands/game/ActivateSensorCommand.java
package commands.game;

import java.util.UUID;

import commands.base.CommandBase;
import communication.MessageInformation;
import exceptions.InvalidModelOperationException;
import models.game.Map;
import models.game.Placeable;
import models.game.PlaceableOnSquare;
import models.game.Rail;
import models.session.RoRSession;

public class ActivateSensorCommand extends CommandBase {

	private UUID selectedModelId;
	
	public ActivateSensorCommand(RoRSession session, MessageInformation messageInfo) {
		super(session, messageInfo);
		this.selectedModelId = messageInfo.getValueAsUUID("selectedModelId");
	}

	@Override
	public void execute() {
		Map map = this.session.getMap();
		Placeable placeable = map.getPlaceableById(selectedModelId);
		if (placeable instanceof Rail) {
			Rail rail = (Rail)placeable;
			rail.activateSensor();
		} else {
			throw new InvalidModelOperationException("Keine Rail");
		}
	}
}

File: RoRClient/Commands/Game/Other/ActivateSensorCommand.cs
using RoRClient.Commands.Base;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using RoRClient.Communication.DataTransferObject;
using RoRClient.Models.Session;
using RoRClient.Models.Game;

namespace RoRClient.Commands.Game.Other
{
    public class ActivateSensorCommand : CommandBase
    {
        private bool active;
        private Guid railId;

        public ActivateSensorCommand(RoRSession session, MessageInformation message) : base(session, message)
        {
            active = message.GetValueAsBool("active");
            railId = message.GetValueAsGuid("railId");
        }

        public override void Execute()
        {
            Rail rail = (Rail)session.Map.GetPlaceableById(railId);
            Sensor sensor = rail.Sensor;
            sensor.Active = true;
        }
    }
}
